ihmc-high-level-behaviors/src/main/java/us/ihmc/behaviors/behaviorTree/BehaviorTreeDefinitionRegistry.java

```diff
@@ -48,10 +46,78 @@ public static Class<?> getNodeDefinitionClass(byte nodeType)
    {
       for (RegistryRecord definitionEntry : DEFINITIONS)
       {
-         if (nodeType == definitionEntry.messageByte())
-            return definitionEntry.typeClass();
+         if (nodeType == definitionEntry.getMessageByte())
+            return definitionEntry.getTypeClass();
+      }
+
+      return null;
+   }
+
+   public static String getInitialName(Class<?> definitionClass)
+   {
+      for (RegistryRecord definitionEntry : DEFINITIONS)
+      {
+         if (definitionClass == definitionEntry.getTypeClass())
+            return definitionEntry.getInitialName();
       }
 
       return null;
    }
+
+   private static class RegistryRecord
+   {
+      private final String initialName;
+      private final Class<?> typeClass;
+      private final byte messageByte;
+
+      private RegistryRecord(Class<?> typeClass, byte messageByte)
+      {
+         this.typeClass = typeClass;
+         this.messageByte = messageByte;
+
+         if (typeClass == BehaviorTreeNodeDefinition.class)
+         {
+            this.initialName = "Basic node";
+         }
+         else
+         {
+            // Convert "PascalCase" to "Sentence case"
+            String initialName = typeClass.getSimpleName();
+            initialName = initialName.replaceFirst("^BehaviorTree", "");
+            initialName = initialName.replaceFirst("Definition$", "");
+            initialName = initialName.replaceFirst("Node$", " Node");
+            initialName = initialName.replaceAll("([a-z])([A-Z]+)", "$1 $2");
+
+            // Undercase words after first
+            String[] words = initialName.split(" ");
+            if (words.length > 1)
+            {
+               StringBuilder modifiedName = new StringBuilder();
+               modifiedName.append(words[0]);
+               for (int i = 1; i < words.length; i++)
+               {
+                  modifiedName.append(" ").append(words[i].toLowerCase());
+               }
+               initialName = modifiedName.toString();
+            }
+
+            this.initialName = initialName;
```

Comment:
Here we generate good initial names for nodes, so we don't have to hardcode those anymore.